unittests/api_tests.cpp

```diff
@@ -3855,4 +3856,33 @@ BOOST_AUTO_TEST_CASE(get_code_hash_tests) { try {
    check("test"_n, 3);
 } FC_LOG_AND_RETHROW() }
 
+// test set_finalizer host function serialization and tester set_finalizers
+BOOST_AUTO_TEST_CASE(set_finalizer_test) { try {
+   validating_tester t;
+
+   t.produce_block();
+
+   // Create producer accounts
+   vector<account_name> producers = {
+      "inita"_n, "initb"_n, "initc"_n, "initd"_n, "inite"_n, "initf"_n, "initg"_n,
+      "inith"_n, "initi"_n, "initj"_n, "initk"_n, "initl"_n, "initm"_n, "initn"_n,
+      "inito"_n, "initp"_n, "initq"_n, "initr"_n, "inits"_n, "initt"_n, "initu"_n
+   };
+
+   t.create_accounts(producers);
+   t.produce_block();
+
+   // activate hotstuff
+   t.set_finalizers(producers);
+   auto block = t.produce_block();
+
+   std::optional<block_header_extension> ext = block->extract_header_extension(hs_finalizer_set_extension::extension_id());
+
+   BOOST_TEST(!!ext);
+   BOOST_TEST(std::get<hs_finalizer_set_extension>(*ext).finalizers.size() == producers.size());
+   BOOST_TEST(std::get<hs_finalizer_set_extension>(*ext).generation == 1);
+   BOOST_TEST(std::get<hs_finalizer_set_extension>(*ext).fthreshold == producers.size() / 3 * 2);
```

Comment:
Can you check `hs_dpos_irreversible_blocknum` is `uint32_t max()` and `hs_block_confirmed` is `uint16_t max()` after hotstuff activation?

Reply:
Good idea.

Reply:
This points out there is more plumbing needed to the tester as it currently does not have a chain_pacemaker wired up.

Reply:
The `qc_chain`, `chain_pacemaker` needs a bit of refactoring so it can be used in `tester`. How about doing that in a diff PR?